evm-adapters/src/sputnik/cheatcodes/cheatcode_handler.rs

```diff
@@ -966,6 +1091,13 @@ impl<'a, 'b, B: Backend, P: PrecompileSet> Handler for CheatcodeStackExecutor<'a
     }
 
     fn log(&mut self, address: H160, topics: Vec<H256>, data: Vec<u8>) -> Result<(), ExitError> {
+        if self.state().trace_enabled {
```

Comment:
One thing we could do is make a `TRACE_ENABLED` an associated constant on the `state()`'s main trait, so that we don't pay the cost at runtime? Maybe would have negligible gains though.

Reply:
ye maybe we can do that at a later point, but right now not worth the effort imo